ansible_base/authentication/utils/authentication.py

```diff
@@ -125,18 +127,117 @@ def determine_username_from_uid_social(**kwargs) -> dict:
+
+    # PROPOSAL FLOW: Authenticator provides email
+    existing_user_match = user_model.objects.filter(email__iexact=normalized_email)
+    user_count = existing_user_match.count()
+
+    # PROPOSAL FLOW: Are there multiple AAP users with this Email? Yes
+    if user_count > 1:
+        logger.warning(f"Found more than 1 user matching {uid}/{email}, unable to determine which user to merge with!")
+        raise AuthException(f"Found more than 1 user matching {uid}/{email}, unable to determine which user to merge with!")
+    # PROPOSAL FLOW: Are there multiple AAP users with this Email? No only 1
+    elif user_count == 1:
+        return existing_user_match.first().username
+    # PROPOSAL FLOW: Does an AAP User with this email exist? No
+    else:
+        new_username = get_local_username({'username': uid})
+        logger.warning(
+            f"Authenticator {authenticator.name} wants to authenticate {uid}/{email} "
+            "but that username is already in use by another authenticator, "
+            f"the user from this authenticator will be {new_username}"
+        )
+        return new_username
+
+
+# def _handle_email_username_fallback_strategy(uid: str, email: Union[str, list[str], None], authenticator: Authenticator) -> str:
+#     """Handles the unused 'Email and Username fallback' strategy."""
+#     # This code is not used anywhere, but is kept here for reference, in case we later want to enable it.
+#     normalized_email = _normalize_and_get_email(email)
+
+#     # Note: The original code uses .get(), which will raise an exception if zero or multiple
+#     # objects are found, and then tries to call .count() on the result, which will fail.
+#     # The logic is preserved here as-is.
+#     if not normalized_email:
+#         existing_user_match = Authenticator.objects.get(user__username=uid)
+#     else:
+#         existing_user_match = Authenticator.objects.get(user__email=normalized_email)
+
+#     if existing_user_match.count() == 1:
+#         existing_user = existing_user_match[0].user
+#         new_username = existing_user.username
+#         logger.info(f"Authenticator {authenticator.name} matched {uid}/{email} from provider {existing_user_match[0].provider.name}, combining users")
+#         return new_username
+#     elif existing_user_match.count() > 1:
+#         raise AuthException(f"Found more than 1 user matching {uid}/{email}, unable to determine which user to merge with!")
+#     else:
+#         # Some other provider is providing this username so we need to create our own username
+#         new_username = get_local_username({'username': uid})
+#         logger.info(
+#             f'Authenticator {authenticator.name} wants to authenticate {uid}/{email} but that'
+#             f' username is already in use by another authenticator,'
+#             f' the user from this authenticator will be {new_username}'
+#         )
+#         return new_username
```

Comment:
should we remove this?

Reply:
This was left on purpose. In conversations earlier with @john-westcott-iv , we noted it would be good to keep this to make potentially revisiting this logic easier later, as we may want to update this in a fast follow.